pkg/cloudprovider/provider/vsphere/provider.go

```diff
@@ -64,11 +66,13 @@ type Config struct {
 	Password         string
 	VSphereURL       string
 	Datacenter       string
+	Cluster          string
```

Comment:
As far as I understand, this field needs to be non-empty when configuring anti-affinity, so can we maybe add such validation to `func (p *provider) Validate`? At best, even query the cluster for existence as it's done with e.g. the datastore cluster.